crypto/src/utils.rs

```diff
@@ -0,0 +1,205 @@
+    let keys: EncryptedKeys =
+        toml::from_slice(file_content.as_slice()).map_err(|e| Error::new(ErrorKind::Other, e))?;
+    keys.decrypt(pass_phrase)
+}
+
+struct PublicKeyHex;
+
+impl Hex<PublicKey> for PublicKeyHex {
+    fn create_bytes(value: &PublicKey) -> Cow<[u8]> {
+        Cow::Borrowed(&*value.as_ref())
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Result<PublicKey, String> {
+        PublicKey::from_slice(bytes)
+            .ok_or_else(|| "Couldn't create PublicKey from slice".to_owned())
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct EncryptedKeys {
+    #[serde(with = "PublicKeyHex")]
+    public_key: PublicKey,
+    secret_key: ErasedPwBox,
+}
+
+impl EncryptedKeys {
+    fn encrypt(
+        public_key: PublicKey,
+        secret_key: &SecretKey,
+        pass_phrase: &[u8],
+    ) -> Result<EncryptedKeys, Error> {
+        let mut rng = thread_rng();
+        let mut eraser = Eraser::new();
+        let seed = &secret_key[..SEED_LENGTH];
+        eraser.add_suite::<Sodium>();
+        let pwbox = Sodium::build_box(&mut rng)
+            .seal(pass_phrase, &seed)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't create a pw box"))?;
+        let encrypted_key = eraser
+            .erase(&pwbox)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't convert a pw box"))?;
+
+        Ok(EncryptedKeys {
+            public_key,
+            secret_key: encrypted_key,
+        })
+    }
+
+    fn decrypt(self, pass_phrase: &[u8]) -> Result<(PublicKey, SecretKey), Error> {
+        let mut eraser = Eraser::new();
+        eraser.add_suite::<Sodium>();
+        let restored = match eraser.restore(&self.secret_key) {
+            Ok(restored) => restored,
+            Err(_) => {
+                return Err(Error::new(
+                    ErrorKind::Other,
+                    "Couldn't restore a secret key",
+                ))
+            }
+        };
+        assert_eq!(restored.len(), SEED_LENGTH);
+        let seed_bytes = restored
+            .open(pass_phrase)
+            .map_err(|_| Error::new(ErrorKind::Other, "Couldn't open an encrypted key"))?
+            .to_vec();
+        let seed = Seed::from_slice(&seed_bytes[..])
+            .ok_or_else(|| Error::new(ErrorKind::Other, "Couldn't create seed from slice"))?;
+        let (public_key, secret_key) = gen_keypair_from_seed(&seed);
+        assert_eq!(self.public_key, public_key);
+        Ok((public_key, secret_key))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    extern crate tempdir;
+
+    use self::tempdir::TempDir;
+    use super::*;
+    use toml;
+
+    #[test]
+    fn test_create_and_read_keys_file() {
+        let dir = TempDir::new("test_utils").expect("Couldn't create TempDir");
+        let file_path = dir.path().join("private_key.toml");
+        let pass_phrase = b"passphrase";
+        let result = create_keys_file(file_path.as_path(), pass_phrase);
+        assert!(result.is_ok());
```

Comment:
Nit: seems redundant given an `unwrap()` on the next line. Generally, I'd rewrite this as

```rust
let pk = create_keys_file(file_path.as_path(), pass_phrase).unwrap();
```

or even

```rust
let pk = create_keys_file(file_path.as_path(), pass_phrase)?;
```

(Custom return types for tests are now supported in stable Rust.)